src/transformers/training_args.py

```diff
@@ -483,3 +495,10 @@ def to_sanitized_dict(self) -> Dict[str, Any]:
             valid_types.append(torch.Tensor)
 
         return {k: v if type(v) in valid_types else str(v) for k, v in d.items()}
+
+
+class DistributedEnvironment(Enum):
+    SINGLE = "single"
+    PARALLEL = "parallel"
+    DISTRIBUTED_PARALLEL = "distributed_parallel"
+    TPU = "tpu"
```

Comment:
perhaps sticking to the commonly used abbreviations: dp/dpp? 

* dp = DataParallel
* ddp = DistributedDataParallel
* tpu = TPU 

and "single" is too ambiguous - single gpu or single process? 

ideas:

* np = NormalProcess
* sp = SimpleProcess
* ss = SinglegpuorcpuSingleprocess
* pu = ProcessingUnit (G or C)

Reply:
We will need a mode for MP (ModelParallel) and PP (PipeParallel) too.

Reply:
A three letter abbreviation won't be clear for users who are not super familiar with PyTorch and we generally try to avoid them in the Transformers codebase.

Reply:
Sure, but there are existing conventions - why reinvent names when we have them already. Can use the full `DistributedDataParallel`, etc.